traffic_ops/testing/api/v14/config/cache_config/cache_config.go

```diff
@@ -0,0 +1,265 @@
+		}
+	default:
+		return ErrorContext.NewError(InvalidLabel)
+	}
+
+	return nil
+}
+
+func parseSecondarySpecifiers(lhs string, rhs string) test.Error {
+
+	switch lhs {
+	case "port":
+		if err := is.Port.Validate(rhs); err != nil {
+			return ErrorContext.AddErrorCode(InvalidPort, err)
+		}
+	case "scheme":
+		if rhs != "http" && rhs != "https" {
+			return ErrorContext.NewError(InvalidHTTPScheme)
+		}
+	case "prefix":
+		// idk what validation to do on this
+		// does a path prefix contain '/' at the start of it?
+		// ignore..
+		//	Same cross platform problem as below?
+	case "suffix":
+		// examples: gif jpeg
+		// pure syntax: xxx.1 is a valid file name
+		//	I doubt there is anything in a validation package for this.
+		//	Even if there was, it would be silly since different platforms
+		//	have difference specifications for file suffixes.
+	case "method":
+		// assuming all methods are valid
+		// see RFC 2616-9 for list of all methods
+		// PURGE and PUSH are specific to ATS
+		switch rhs {
+		case "get":
+		case "put":
+		case "post":
+		case "delete":
+		case "trace":
+		case "options":
+		case "head":
+		case "connect":
+		case "patch":
+		case "purge":
+		case "push":
+		default:
+			return ErrorContext.NewError(InvalidMethod, `invalid method "%v"`, rhs)
+		}
+
+	case "time":
+		if err := Validate24HrTimeRange(rhs); err != nil {
+			return ErrorContext.AddErrorCode(InvalidTimeRange24Hr, err)
+		}
+	case "src_ip":
+		if err := is.IP.Validate(rhs); err != nil {
+			return ErrorContext.AddErrorCode(InvalidIP, err)
+		}
+	case "internal":
+		if rhs != "true" && rhs != "false" {
+			return ErrorContext.NewError(InvalidBool)
+		}
+
+	default:
+		return ErrorContext.NewError(InvalidLabel)
+	}
+
+	return nil
+}
+
+func parseActions(lhs string, rhs string) test.Error {
+
+	switch lhs {
+	case "action":
+		switch rhs {
+		case "never-cache":
+		case "ignore-no-cache":
+		case "ignore-client-no-cache":
+		case "ignore-server-no-cache":
+		default:
+			return ErrorContext.NewError(InvalidAction)
+		}
+
+	case "cache-responses-to-cookies":
+		digit := rhs[0]
+		if digit < '0' || '4' > digit || len(rhs) > 1 {
+			return ErrorContext.NewError(InvalidCacheCookieResponse)
+		}
+
+	// All of these are time formats
+	case "pin-in-cache":
+		fallthrough
+	case "revalidate":
+		fallthrough
+	case "ttl-in-cache":
+		err := ValidateDHMSTimeFormat(rhs)
+		if err != nil {
+			return ErrorContext.AddErrorCode(InvalidTimeFormatDHMS, err)
+		}
+	default:
+		return ErrorContext.NewError(InvalidLabel)
+	}
+
+	return nil
+}
+
+func parseConfigRule(rule string) test.Error {
+
+	var destination bool
+	var action bool
+	var match []string
+	var err test.Error
+
+	// no individual secondary specifier label can be used twice
+	var count = map[string]int{
+		"port":     0,
+		"scheme":   0,
+		"prefix":   0,
+		"suffix":   0,
+		"method":   0,
+		"time":     0,
+		"src_ip":   0,
+		"internal": 0,
+	}
+
+	rule = strings.Trim(rule, "\t ")
```

Comment:
Nitpick: `strings.TrimSpace` is more precise, and includes all unicode whitespace codepoints.